### DISCUSSION TITLE
Include Parameter Identification results in JSON export

### DISCUSSION THREAD

**Discussion by Erik-Sjogren** [Open-Systems-Pharmacology/Forum#1099] [Questions & Problems]
Hi,
Is there a possibility to include the results from parameter identifications in the JSON export?
This seems not to be done automatically.
Would be very valuable.
Many thanks

**Comment by msevestre** [comment]
Hi @Erik-Sjogren 
What json export are you talking about?

**Comment by Erik-Sjogren** [comment]
Sorry, 
I mean by using the functionality "Export to Snapshot"

**Reply by Erik-Sjogren** [reply]
Subheading "Export project to snapshot.."

**Comment by msevestre** [comment]
Hi @Erik-Sjogren 

What would be the reason to include the "results" of a PI? With Results do you mean the optimized parameters and/or the plots associated with a PI

The goal behind the snapshot is to be able to recreate a project from scratch using only the parameters that were changed by the user (input parameters) so that the project is "fresh" so to speak. There is no guarantee for instance that running the same PI will lead to the same results as some model internals might have changed etc...

We also did not want to run all the PI by default as this would take a lot of time to load a snapshot.

**Reply by Erik-Sjogren** [reply]
Hi @msevestre 
In reporting it is important to verify/trace that the used/reported parameters actually are those that were "identified" in model development. So for me the parameters (+confidence) are the most important. 
Currently (correct me if I am wrong) one needs to actively access the project file (e.g. *.pksim5) and manually, via the interface, locate the parameters (and confidence) in respective "parameter identification".  
A possibility to automate this process, for instance script based via R, would be a big improvement. It looks like the JSON structure can accommodate for this.  

I absolutely agree with you that the PIs shouldn't re-run at import. Absolutely not... that would be very bad.

**Reply by msevestre** [reply]
Oh I see. You are using the snapshot file as some kind of reporting tool.... interesting. We'll discuss this internally. The snapshot is mostly input and not output. Maybe it makes sense to export this, or create another type of export with the snapshot and results...

**Reply by Erik-Sjogren** [reply]
Many thanks @msevestre 
I was thinking it could be done similarly as for the simulations, for which output and info about observations is included in the JSON.

**Reply by Erik-Sjogren** [reply]
And also including the results from the PI.

**Reply by msevestre** [reply]
@Erik-Sjogren For a simulation we save which outputs are selected (input) but not the output values (results of calculations). That's why this feels like something different.

**Reply by sfrechen** [reply]
Agree with @msevestre that the snaphot functionality was not intended to save any previous simulation results which includes obviously parameter identifcation results.
On the other, I also sometimes missed that "feature" @Erik-Sjogren is asking for. In MoBi, I think we have (or had?) kind of a history for simulation results. Maybe one can think of something like this for PIs (not for simulations!) which saves date and PK-Sim version (and is cleared in case the configuration of the PI changes). Such a history could then become part of the snaphot...

**Reply by msevestre** [reply]
>Such a history could then become part of the snaphot...

How would we use it in the snapshot?

**Reply by Erik-Sjogren** [reply]
Agree, a possibility to extract the time stamp for the PI would also be good.
This info would however be quite redundant if you were not able to extract the parameters themselves.
